### PR TITLE
fix: Posthog env variable fix

### LINKED ISSUE #1440: Bug: Posthog not sending analytics data
### Describe the bug

Open sauced will not send analytics data to posthog once deployed.

### Steps to reproduce

Run the application, click on the login button and then click on both onboarding butons in the onboarding flow.

### Affected services

opensauced.pizza

### Platforms

_No response_

### Browsers

_No response_

### Environment

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by chadstewart**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR [fixes] the [bug] of posthog not sending analytics data.. 

## Related Tickets & Documents

Fixes #1440 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by chadstewart** [comment]
So I made some changes to the analytics file to use import.meta because that's how Vite allows the app to use env variables. The problems seems to be that the jest tests fail now so now I'll need to update the tests to get them to pass.

Any ideas on what to do here?

**Comment by chadstewart** [comment]
Okay, so I came up with a bit of a work around but it involved messing with vite.config.ts which I am not comfortable with modifying. Could anyone take a look at my last commit and let me know if what I did was a good idea?

File: src/lib/analytics.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import posthog from "posthog-js";
 
 function initiatePostHog () {
-    posthog.init(process.env.POSTHOG_ID, { api_host: 'https://app.posthog.com' });
+    posthog.init(process.env.VITE_POSTHOG_ID, { api_host: 'https://app.posthog.com' });
```
Comment (bdougie) [inline src/lib/analytics.js:4]: Per this [PR](https://github.com/open-sauced/open-sauced/pull/1442/files#diff-a96ec4e9261995cfb834805e073040556daa2d946f40771a73c228f2f5589004R4), it should be `import.meta.VITE_POSTHOG_ID`

**Comment by 0-vortex** [comment]
I saw the ticket wasn't `.take`n so I figured @chadstewart could use some support with this in https://github.com/open-sauced/open-sauced/pull/1442

**Comment by 0-vortex** [comment]
> Thanks for taking this on! If you have not already, join the conversation in our [Discord](https://discord.gg/U2peSNf23P)

and this is a bug :D

**Comment by chadstewart** [comment]
> I saw the ticket wasn't `.take`n so I figured @chadstewart could use some support with this in #1442

Honestly, when I made the ticket I forgot to assign myself to it. 🤣

File: src/lib/analytics.js (hunk 1 above)
Comment (bdougie) [reply src/lib/analytics.js:4]: Nvm, but the resolution is allow listing the urls in posthog and exposing the id

**Comment by chadstewart** [comment]
No longer needed because #1442 solves this issue.

**Pull request closed without merging**